src/managers/backup.py

```diff
@@ -136,6 +137,26 @@ def create_backup(self) -> BackupMetadata:
 
         return metadata
 
+    def list_backups(self) -> list[BackupMetadata]:
+        """List snapshots present in object storage."""
+        backups_metadata: list[BackupMetadata] = []
+
+        # S3 API is limited so we end up with an N+1 query problem. Thus, we limit the
+        # number of backups and fetch the latest modified ones.
+        remote_files = self.bucket.objects.filter(Prefix=self.backups_path)
+        remote_metadata = filter(lambda rfile: rfile.key.endswith(".yaml"), remote_files)
+        sorted_remote_files = sorted(
+            remote_metadata, key=attrgetter("last_modified"), reverse=True
+        )
+
+        for remote_file in islice(sorted_remote_files, S3_BACKUPS_LIMIT):
+            buffer = BytesIO()
+            self.bucket.download_fileobj(remote_file.key, buffer)
```

Comment:
**suggestion**: I only glanced ad the docs, but I'm not sure how errors on connecting to the endpoint are handled. Might be worth adding retries from the lib itself to the initialization of the bucket maybe. `wait_until_exists` will raise after ~2 minutes for example.

Reply:
Would you be ok if I added this change in the upcoming restore PR to avoid re-opening on VM? The change is to pass a global `config=Config(retries={"mode": "standard", "max_attempts": 3})` parameter to the s3 resource